rest-api-spec/src/main/resources/rest-api-spec/test/search/320_disallow_queries.yml

```diff
@@ -0,0 +1,330 @@
+---
+setup:
+  - skip:
+      version: " - 7.99.99"
+      reason: "implemented in 8.0.0"
+
+  - do:
+      indices.create:
+        index: test
+        body:
+          mappings:
+            properties:
+              text:
+                type: text
+                analyzer: standard
+                fields:
+                  raw:
+                    type: keyword
+              nested1:
+                type: nested
+
+  - do:
+      bulk:
+        refresh: true
+        body:
+          - '{"index": {"_index": "test", "_id": "1"}}'
+          - '{"text" : "Some like it hot, some like it cold", "nested1": [{"foo": "bar1"}]}'
+          - '{"index": {"_index": "test", "_id": "2"}}'
+          - '{"text" : "Its cold outside, theres no kind of atmosphere", "nested1": [{"foo": "bar2"}]}'
+          - '{"index": {"_index": "test", "_id": "3"}}'
+          - '{"text" : "Baby its cold there outside", "nested1": [{"foo": "bar3"}]}'
+          - '{"index": {"_index": "test", "_id": "4"}}'
+          - '{"text" : "Outside it is cold and wet", "nested1": [{"foo": "bar4"}]}'
+
+---
+teardown:
+  - skip:
+      version: " - 7.99.99"
+      reason: "implemented in 8.0.0"
+
+  - do:
+      cluster.put_settings:
+        body:
+          transient:
+            search.allow_expensive_queries: null
+
+---
+"Test disallow expensive queries":
+  - skip:
+      version: " - 7.99.99"
+      reason: "implemented in 8.0.0"
+
+  ### Check for initial setting = null -> false
+  - do:
+      cluster.get_settings:
+        flat_settings: true
+
+  - match: {search.allow_expensive_queries: null}
```

Comment:
heya @matriv I think this assertion is problematic, it works with the java runner due to the semantics of HashMap.get , but it may fail with test runners written in other languages. I was chatting to @karmi about this and the proper way to do this null check would be is_false: search.allow_expensive_queries . Would you mind please changing this?

Reply:
To clarify, this fails in the Go client runner, what works is `is_false: { search.allow_expensive_queries: null }`.

Reply:
does is_false accept values? I thought the right syntax would be `is_false: search.allow_expensive_queries`

Reply:
`is_false: search.allow_expensive_queries` works with Java.

Reply:
My bad, sorry — `is_false: search.allow_expensive_queries` is correct.